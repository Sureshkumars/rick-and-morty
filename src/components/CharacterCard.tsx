import React from "react";
import { Box, Image, Badge } from "@chakra-ui/core";
import { Character } from "../types/character";
import { Link } from "react-router-dom";

interface IProps {
  character: Character;
}
export const CharacterCard: React.FC<IProps> = ({ character }: IProps) => {
  return (
    <Link to={`/character/${character.characterId}`} title={character.name}>
      <Box
        maxW="280px"
        borderWidth="1px"
        borderRadius="lg"
        overflow="hidden"
        margin="1rem"
      >
        <Image
          src={character.imageUrl}
          alt={character.name}
          fallbackSrc="/character-fallback.jpeg"
        />
        <Box p="6">
          <Box d="flex" alignItems="baseline">
            <Badge borderRadius="full" px="2" colorScheme="teal">
              {character.species}
            </Badge>
          </Box>
          <Box mt="1" as="h4" lineHeight="tight" isTruncated>
            <b>{character.name} </b>
          </Box>
          <Box as="p">
            <b>Last location - </b> {character.lastLocation}
          </Box>
          <Box d="flex" mt="2" alignItems="center">
            <Box as="span" color="gray.600" fontSize="sm">
              Totally {character.totalEpisodes} episodes
            </Box>
          </Box>
        </Box>
      </Box>
    </Link>
  );
};
