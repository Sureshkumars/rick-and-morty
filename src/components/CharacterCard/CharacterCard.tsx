import React from "react";
import { Box, Image, Badge } from "@chakra-ui/core";
import { Character } from "../../types/character";

interface IProps {
  character: Character;
}
export const CharacterCard: React.FC<IProps> = ({ character }: IProps) => {
  return (
    <Box
      maxW="280px"
      borderWidth="1px"
      borderRadius="lg"
      overflow="hidden"
      margin="1rem"
    >
      <Image
        src={character.imageUrl}
        alt={character.name}
        fallbackSrc="/images/character-fallback.jpeg"
      />
      <Box p="6">
        <Box d="flex" alignItems="baseline">
          <Badge borderRadius="full" px="2" colorScheme="teal">
            {character.species}
          </Badge>
        </Box>
        <Box mt="1" as="h4" lineHeight="tight" isTruncated>
          <strong>{character.name} </strong>
        </Box>
        <Box as="p">
          <strong>Last location - </strong> {character.lastLocation}
        </Box>
        <Box d="flex" mt="2" alignItems="center">
          <Box as="span" color="gray.600" fontSize="sm">
            Totally {character.totalEpisodes} episodes
          </Box>
        </Box>
      </Box>
    </Box>
  );
};
