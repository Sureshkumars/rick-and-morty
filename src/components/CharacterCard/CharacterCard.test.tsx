import React from "react";
import { CharacterCard } from "./CharacterCard";
import { render } from "@testing-library/react";

const character = {
  name: "Rick",
  imageUrl: "/images/character-fallback.jpeg",
  species: "Human",
  lastLocation: "Earth",
  totalEpisodes: 20,
  characterId: 1,
};

it("Check for Character Card snapshot", () => {
  const { asFragment }: any = render(<CharacterCard character={character} />);
  expect(asFragment()).toMatchSnapshot();
});
