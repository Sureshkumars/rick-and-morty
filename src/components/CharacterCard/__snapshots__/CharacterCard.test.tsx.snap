// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check for Character Card snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-ao3w3m"
  >
    <img
      alt="Rick"
      class="chakra-image__placeholder css-0"
      src="/images/character-fallback.jpeg"
    />
    <div
      class="css-n16x53"
    >
      <div
        class="css-x1sij0"
      >
        <span
          class="chakra-badge css-637omx"
        >
          Human
        </span>
      </div>
      <h4
        class="css-mvaft7"
      >
        <strong>
          Rick 
        </strong>
      </h4>
      <p
        class="css-0"
      >
        <strong>
          Last location - 
        </strong>
         Earth
      </p>
      <div
        class="css-15yujhg"
      >
        <span
          class="css-bgvdb6"
        >
          Totally 20 episodes
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
